src/index.js

```diff
@@ -80,4 +98,52 @@ function render() {
     arToolkitContext.update(arToolkitSource.domElement);
     scene.visible = camera.visible;
   }
+
+  if (ready) {
+    updatePosition();
+    mixer.update(clock.getDelta());
+  }
+}
+
+let xv = 0.02;
+let zv = 0.01;
+
+function updatePosition() {
+  /*
+      If we are within .1 of -.5 for any then start adding
+     */
+  if (model.position.x < -0.4) {
+    xv += 0.001;
+  }
+  if (model.position.z < -0.4) {
+    zv += 0.001;
+  }
+  if (model.position.x > 0.4) {
+    xv -= 0.001;
+  }
+  if (model.position.z > 0.4) {
+    zv -= 0.001;
+  }
+
+  xv += Math.random() * (0.001 - -0.001) + -0.001;
```

Comment:
Mind simplifying the math a bit here? Maybe something like this if you want to maintain the `0.001`:
```suggestion
  xv += Math.random() * (0.001 * 2) - 0.001;
```

Reply:
Done. Was just trying to make the range super clear but if you have half a brain it's pretty clear either way.